sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12-file-detail.ts

```diff
@@ -37,50 +37,79 @@ export class IER12FileDetail implements IER12FileLine {
-    Object.keys(awardsSum).forEach((award: string) => {
-      record.append(award, 4);
+    this.disbursementSchedules.forEach((disbursement) => {
+      record.appendWithStartFiller(this.assessmentId, 10, NUMBER_FILLER);
+      record.appendWithStartFiller(disbursement.id, 10, NUMBER_FILLER);
+      record.appendWithStartFiller(this.applicationNumber, 10, NUMBER_FILLER);
+      record.append(this.sin, 9);
+      record.appendWithEndFiller(this.studentLastName, 25, SPACE_FILLER);
+      record.appendWithEndFiller(this.studentGivenName ?? "", 15, SPACE_FILLER);
+      record.appendDate(this.birthDate, DATE_FORMAT);
+      record.appendWithEndFiller(this.programName, 25, SPACE_FILLER);
+      record.appendWithEndFiller(this.programDescription, 50, SPACE_FILLER);
+      record.appendWithEndFiller(this.credentialType, 25, SPACE_FILLER);
+      record.append(this.cipCode.toString(), 6);
+      record.append(this.nocCode, 4);
+      record.appendWithEndFiller(this.sabcCode ?? "", 4, SPACE_FILLER);
+      record.appendWithEndFiller(
+        this.institutionProgramCode ?? "",
+        25,
+        SPACE_FILLER,
+      );
+      record.append(this.programLength.toString(), 1);
+      record.appendDate(this.studyStartDate, DATE_FORMAT);
+      record.appendDate(this.studyEndDate, DATE_FORMAT);
+      record.appendWithStartFiller(
+        this.tuitionFees.toString(),
+        8,
+        NUMBER_FILLER,
+      );
+      record.appendWithStartFiller(
+        this.programRelatedCosts.toString(),
+        8,
+        NUMBER_FILLER,
+      );
       record.appendWithStartFiller(
-        awardsSum[award].toString(),
+        this.mandatoryFees.toString(),
         8,
         NUMBER_FILLER,
       );
+      record.appendWithStartFiller(
+        this.exceptionExpenses.toString(),
+        8,
+        NUMBER_FILLER,
+      );
+      record.appendWithStartFiller(
+        this.totalFundedWeeks.toString(),
+        2,
+        NUMBER_FILLER,
+      );
+      record.repeatAppend(SPACE_FILLER, 3); //We have hardcoded the courseLoad to null as its only for FullTime.
+      record.append("F", 1); //This implementation is only for FullTime, so hardcoding it to F.
+      record.appendWithEndFiller(
+        disbursement.coeStatus ?? "",
+        10,
+        SPACE_FILLER,
+      );
+      record.appendWithEndFiller(
+        disbursement.disbursementScheduleStatus ?? "",
+        10,
+        SPACE_FILLER,
+      );
+      record.appendDate(disbursement.disbursementDate, DATE_FORMAT);
+      record.appendDate(disbursement.dateSent, DATE_FORMAT);
+      disbursement.disbursementValues.forEach(
+        (disbursementValue: DisbursementValue) => {
+          record.append(disbursementValue.valueCode, 4);
+          record.appendWithStartFiller(
+            disbursementValue.valueAmount,
+            8,
+            NUMBER_FILLER,
+          );
+        },
+      );
+      record.append("\r\n");
```

Comment:
I believe that this will cause the `\r\n` to be added twice. The append for the last record must not happen.
The `for` can generate an array of string and the return can just do a join as below:
```ts
return records.join("\r\n");
```
Also, it would be good to create a const to be used here and in the `sources\packages\backend\libs\integrations\src\services\ssh\sftp-integration-base.ts`